slack/decorators/workflow.py

```diff
@@ -0,0 +1,59 @@
+import logging
+logger = logging.getLogger(name="statuspage init")
+
+from django.db.models.signals import post_save, post_migrate
+from django.db.utils import ProgrammingError
+from django.dispatch import receiver
+
+from slack.apps import SlackConfig
+from slack.models import Workflow, WorkflowParameters, CommsChannel, workflow_post_save
+
+RESPONSE_WORKFLOWS = {}
+
+
+def register_workflow(name, *args, **kwargs):
+    def _wrapper(wf):
+        imp = wf(*args, **kwargs)
+        RESPONSE_WORKFLOWS[name] = imp
+        try:
+            wf, created = Workflow.objects.get_or_create(name=name)
+            if wf.enabled:
+                imp.enabled(wf)
+            else:
+                imp.disabled(wf)
+            return wf
+        except ProgrammingError as ex: # Throws errors before migration can be completed (first run)
+            logger.error(ex)
+    return _wrapper
+
+
+@receiver(post_migrate, dispatch_uid='setup_workflows') # Ensure everything is setup after migration
+def setup_workflows(sender, **kwargs):
+    if not sender.name == 'slack':
+        return
+    for name, workflow in RESPONSE_WORKFLOWS.copy().items():
+        wf, created = Workflow.objects.get_or_create(name=name)
+        if created:
+            wf.save()
+        for wfp in workflow.configuration_values:
+            cnf, created = WorkflowParameters.objects.get_or_create(name=wfp, workflow=wf)
+        if wf.enabled:
+            workflow.enabled(wf)
+        else:
+            workflow.disabled(wf)
+
+@receiver(workflow_post_save,  dispatch_uid="update_workflow") # Update after admin changes
+def update_workflow(sender, workflow, **kwargs):
+    if workflow.enabled:
+        RESPONSE_WORKFLOWS[workflow.name].enabled(workflow)
+    else:
+        RESPONSE_WORKFLOWS[workflow.name].disabled(workflow)
+
+
+class WorkFlowClass(object):
```

Comment:
Just for consistency
```suggestion
class WorkflowClass(object):
```